src/data-types/varbinary.ts

```diff
@@ -56,14 +56,43 @@ const VarBinary: { maximumLength: number } & DataType = {
   },
 
   generate: function* (parameter, options) {
-    if (parameter.value != null) {
-      const buffer = new WritableTrackingBuffer(0);
+    let value = parameter.value;
+
+    if (value != null) {
       if (parameter.length! <= this.maximumLength) {
-        buffer.writeUsVarbyte(parameter.value);
-      } else {
-        buffer.writePLPBody(parameter.value);
+        const buffer = new WritableTrackingBuffer(0);
+        buffer.writeUsVarbyte(value);
+        yield buffer.data;
+
+      } else { // writePLPBody
+        const UNKNOWN_PLP_LEN = Buffer.from([0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff]);
+
+        let length;
+        if (value instanceof Buffer) {
+          length = value.length;
+        } else {
+          value = value.toString();
+          length = Buffer.byteLength(value, 'ucs2');
+        }
+
+        let buffer = Buffer.alloc(4);
+        if (length > 0) {
+          buffer.writeUInt32LE(length, 0);
+
+          if (value instanceof Buffer) {
+            buffer = Buffer.concat([buffer, value], buffer.length + value.length);
+
+          } else {
+            const buffer2 = Buffer.from(value, 'ucs2');
+            buffer = Buffer.concat([buffer, buffer2], buffer.length + buffer2.length);
+          }
+        }
+
+        const end = Buffer.from([0x00, 0x00, 0x00, 0x00]);
+        yield Buffer.concat([UNKNOWN_PLP_LEN, buffer, end], UNKNOWN_PLP_LEN.length + buffer.length + end.length);
```

Comment:
Whenever `Buffer.concat` is called, a new `Buffer` gets allocated with a size that equals the length of all passed in buffers summed together, and then data is copied over from the passed in buffers into the new result buffer.

You can improve both performance as well as memory usage here by not doing that, and instead `yield`ing each chunk that you passed to `Buffer.concat` instead:

```js
        if (length > 0) {
          let buffer = Buffer.alloc(4);
          buffer.writeUInt32LE(length, 0);
          yield buffer;

          if (value instanceof Buffer) {
            yield value;
          } else {
            yield Buffer.from(value, 'ucs2');
          }
        }

        const end = Buffer.from([0x00, 0x00, 0x00, 0x00]);
        yield end;
```

Reply:
Oh yup, varbinary jumped from **16.49 ops/sec** to **6,594.82 ops/sec** 😅